packages/ui/molecules/Menu/Menu.js

```diff
@@ -0,0 +1,241 @@
+import { Base } from '@studiometa/js-toolkit';
+import { nextTick } from '@studiometa/js-toolkit/utils';
+import MenuBtn from './MenuBtn.js';
+import MenuItems from './MenuItems.js';
+
+/**
+ * @typedef {Menu & {
+ *   $children: {
+ *     Menu: Menu[],
+ *     MenuBtn: MenuBtn[],
+ *     MenuItems: MenuItems[],
+ *   }
+ * }} MenuInterface
+ */
+
+/**
+ * Menu class.
+ */
+export default class Menu extends Base {
+  /**
+   * Config.
+   */
+  static config = {
+    name: 'Menu',
+    debug: true,
+    components: {
+      MenuBtn,
+      MenuItems,
+      Menu,
+    },
+    options: {
+      mode: {
+        type: String,
+        default: 'click', // or 'hover'
+      },
+    },
+  };
+
+  /**
+   * Get direct children.
+   *
+   * @this    {MenuInterface}
+   * @param   {string} name
+   * @returns {any[]}
+   */
+  getDirectChildren(name) {
+    if (!this.$children[name]) {
+      return [];
+    }
+
+    if (!this.$children.Menu) {
+      return this.$children[name];
+    }
+
+    return this.$children[name].filter((child) =>
+      this.$children.Menu.every((menu) =>
+        menu.$children[name] ? !menu.$children[name].includes(child) : true
+      )
+    );
+  }
+
+  /**
+   * Get the first `MenuItems` instance.
+   *
+   * @this    {MenuInterface}
+   * @returns {MenuItems}
+   */
+  get menuItems() {
+    return this.getDirectChildren('MenuItems')[0];
+  }
+
+  /**
+   * Get the first `MenuBtn` instance.
+   *
+   * @this    {MenuInterface}
+   * @returns {MenuBtn}
+   */
+  get menuBtn() {
+    return this.getDirectChildren('MenuBtn')[0];
+  }
+
+  /**
+   * @todo test breakpoint to switch between click and hover.
+   */
+  get shouldReactOnClick() {
+    return this.$options.mode === 'click';
+  }
+
+  /**
+   * Wether the button or the items are hovered.
+   *
+   * @returns {boolean}
+   */
+  get isHover() {
+    return this.menuBtn.isHover || this.menuItems.isHover;
+  }
+
+  /**
+   * Set attributes on mounted, destroy the component if it is missing required
+   * child components.
+   *
+   * @returns {void}
+   */
+  mounted() {
+    if (!this.menuBtn || !this.menuItems) {
+      this.$destroy();
```

Comment:
**improve** Maybe throw an exception `The required child components menuBtn or menuItems is missing`

Reply:
I was tempted to do that, but that might be risky if somehow there is a missing part in the template introduced by a user. I preferred a silent `$destroy()` to avoid potentially blocking other scripts with an error.